completions/ssh

```diff
@@ -246,7 +246,7 @@ _ssh_identityfile()
     [[ -z $cur && -d ~/.ssh ]] && cur=~/.ssh/id
     _filedir
     if ((${#COMPREPLY[@]} > 0)); then
-        COMPREPLY=($(compgen -W '${COMPREPLY[@]}' \
+        COMPREPLY=($(compgen -W '${COMPREPLY[@]+"{COMPREPLY[@]}"}' \
```

Comment:
IIRC I've guarded against this in existing code by an explicit check on the number of elements in the accessed array, such as is visible in this particular change here, a few lines above. I don't have that strong opinions how to check it, but I think it would be good to be consistent.

But re this particular change, I think it's unnecessary, as the array access is within the said guarding `if` block already.

Reply:
You are right. I actually replaced some of these `-W '${COMPREPLY[@]}'` just by the regular expression I will again go through the changes to see if they are needed one by one.

Anyway, these changes are yet not the complete list for the `compgen -W '${arr[@]}'` pattern, and I locally have additional changes and still looking for remaining. It seems like this will be another big change, now I'm thinking of separating the PR.

Reply:
Oh, I have been misreading your comment. Now I'm going to rewrite them to use explicit checks.

Reply:
There are so many related fixes, so I decided to separate the PR for `-W '${COMPREPLY[@]}'` at #559. Now in this PR, I don't include the fixes for `-W '${COMPREPLY[@]}'`